### DISCUSSION TITLE
Nested fragments with cache.writeFragment

### DISCUSSION THREAD

**Discussion by schulzfel** [urql-graphql/urql#1940] [Q&A] [answered]
I'm trying to update a nested fragment using writeFragment but it doesn't update. Is this the expected behaviour?
How can I update the value on voted: { value: ...} ?

[code](https://gist.github.com/schulzf/bfb615d854ee9ac39f96fe3d7d60026f) (the markdown was tricky here sorry)

Here is the output of the console.logs:

`previous:  {
  "_id": "6139cf4b30f57a73de850101",
  "points": -1,
  "voted": {
    "value": -1,
    "__typename": "Vote"
  }
}`

`
after:  {
  "_id": "6139cf4b30f57a73de850101",
  "points": 1,
  "voted": {
    "value": -1,
    "__typename": "Vote"
  }
}
`

**Comment by kitten** [comment, marked as answer]
I noticed you're attempting to write: `voted: { value },`, which means it's missing a `__typename` field. You're probably seeing a related console warning for this as well 😃 Once you add the type name for the `voted` field it'll work just fine

**Reply by schulzfel** [reply]
Like you said, added __typename and it works as expected.

thank you for your fast reply! 🙏
